saleor/graphql/attribute/types.py

```diff
@@ -656,3 +655,791 @@ class AttributeValueInput(BaseInputObjectType):
+        if not root.values:
+            return None
+        attr_value = root.values[0].node
+        return CategoryByIdLoader(info.context).load(attr_value.reference_category_id)
+
+
+class AssignedSingleCollectionReferenceAttribute(BaseObjectType):
+    value = graphene.Field(
+        "saleor.graphql.product.types.Collection",
+        description="The assigned collection reference.",
+        required=False,
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents single collection reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData, info: ResolveInfo
+    ) -> Promise[ChannelContext[product_models.Collection]] | None:
+        if not root.values:
+            return None
+
+        channel_slug = root.attribute.channel_slug
+        attr_value = root.values[0].node
+
+        return (
+            CollectionByIdLoader(info.context)
+            .load(attr_value.reference_collection_id)
+            .then(
+                lambda collection: ChannelContext(
+                    node=collection, channel_slug=channel_slug
+                )
+            )
+        )
+
+
+class AssignedMultiPageReferenceAttribute(BaseObjectType):
+    value = NonNullList(
+        "saleor.graphql.page.types.Page",
+        description="List of assigned page references.",
+        required=True,
+        limit=PositiveInt(
+            description=(
+                "Maximum number of referenced pages to return. "
+                f"Default is {DEFAULT_NESTED_LIST_LIMIT}."
+            ),
+            default_value=DEFAULT_NESTED_LIST_LIMIT,
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents multi page reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[ChannelContext[page_models.Page]]]:
+        if not root.values:
+            return Promise.resolve([])
+
+        channel_slug = root.attribute.channel_slug
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        def _wrap_with_channel_context(
+            pages: list[page_models.Page],
+        ) -> list[ChannelContext[page_models.Page]]:
+            if not pages:
+                return []
+            return [
+                ChannelContext(node=page, channel_slug=channel_slug) for page in pages
+            ]
+
+        return (
+            PageByIdLoader(info.context)
+            .load_many([value.reference_page_id for value in attr_values])
+            .then(_wrap_with_channel_context)
+        )
+
+
+class AssignedMultiProductReferenceAttribute(BaseObjectType):
+    value = NonNullList(
```

Comment:
In this field, Saleor returns multiple values. 
```suggestion
    values = NonNullList(
```